MWSE/NIPixelData.cpp

```diff
@@ -43,6 +43,77 @@ namespace NI {
 		return widths[mipMapLevel];
 	}
 
+	void PixelData::exportTGA(const char* fileName) const {
+		if (pixelFormat.format != PixelFormat::Format::RGBA) {
+			std::exception("Unsupported pixel format for export to TGA.");
+		}
+		HANDLE hFile = CreateFileA(
+			fileName,
+			GENERIC_WRITE,
+			0,
+			NULL,
+			CREATE_ALWAYS,
+			FILE_ATTRIBUTE_NORMAL,
+			NULL
+		);
+		if (hFile == INVALID_HANDLE_VALUE) {
+			std::exception("Couldn't open the file for export.");
+		}
+		auto width = getWidth();
+		auto height = getHeight();
+		HeaderTGA header{
+			0, // idLength - No image description
+			0, // colormapType - No color pallete
+			HeaderTGA::DataType::UNCOMPRESSED_RGB,
+			0, // colormapOrigin
+			0, // colormapLength
+			0, // colormapDepth
+			0, // xOrigin
+			0, // yOrigin
+			static_cast<short>(width),
+			static_cast<short>(height),
+			static_cast<unsigned char>(bytesPerPixel * 8), // bitsPerPixel
+			0x08 | 0x20 // Flag bit 3: 8 bits for alpha + flag bit 5: origin in the upper left-hand corner
+		};
+
+		// Write the TGA header
+		DWORD bytesWritten = 0;
+		WriteFile(
+			hFile,
+			&header,
+			sizeof(header),
+			&bytesWritten,
+			NULL
+		);
```

Comment:
I am not checking the bytesWritten after writing, assuming the write operation won't fail. Should I perform a check instead?

Reply:
You can check it, but it's kind of edge-casey. If you do error, be sure to close the file handle before doing so.

Reply:
Okay, I made it check the return value of `WriteFile` and the `bytesWritten`.